SSL cache part 2

pluknet opened the pull request.

pluknet commented:
@bavshin-f5 applied, tnx

pluknet commented:
Updated on top of nginx/nginx@b16199d.

pluknet commented:
- skip a minor test for ssl_cache_object_inheritance on win32
- regrouped to run the main inheritance test first

bavshin-f5 reviewed (approved):
A possible scenario that is not tested is replacing unencrypted key with an encrypted one and verifying that we haven't accidentally cached it after revalidation (https://github.com/nginx/nginx/pull/287#discussion_r1901391209).

Other than that and a couple of cosmetic nits, looks good!

pluknet commented on ssl_cache_reload.t:
```diff
@@ -0,0 +1,139 @@
+
+	$t->reload();
+
+	for (1 .. 10) {
```
I'm fine to increase it as suggested. This won't affect execution time for tests being passed.

The value was chosen at that time pretty arbitrary, based on the assumption that reload time should not exceed some value, and so far it is a good safe default.

pluknet commented on ssl_cache_reload.t:
```diff
@@ -0,0 +1,139 @@
+		or die "Can't create certificate for $name: $!\n";
+}
+
+$t->try_run('no ssl_object_cache_inherit')->plan(5);
```
For sure, there are might be more uncovered cases, as reported by LCOV - code coverage report.
This is an obvious one, thanks for reminding.

pluknet commented:
- addressed comments
- sprinkled comments and style
- added test for encrypted certificate keys to stream proxy, for consistency (see also nginx/nginx#458)
